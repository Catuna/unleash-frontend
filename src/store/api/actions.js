import api from '../../data/api';

export const RECIEVE_API_DETAILS = 'RECIEVE_API_DETAILS';
export const ERROR_RECIEVE_API_DETAILS = 'ERROR_RECIEVE_API_DETAILS';

export const RECEIVE_APPLICATION = 'RECEIVE_APPLICATION';

const recieveApiDetails = json => ({
    type: RECIEVE_API_DETAILS,
    value: json,
});

const errorRecieveApiDetails = (statusCode, type = ERROR_RECIEVE_API_DETAILS) => ({
    type,
    statusCode,
});

export function fetchAll() {
    return dispatch =>
        api
            .fetchAll()
            .then(json => dispatch(recieveApiDetails(json)))
            .catch(error => dispatch(errorRecieveApiDetails(error)));
}
