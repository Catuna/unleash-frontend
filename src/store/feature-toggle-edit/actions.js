const debug = require('debug')('unleash:feature-toggle-edit-actions');

export const TOGGLE_EDIT_INIT = 'TOGGLE_EDIT_INIT';
export const TOGGLE_EDIT_CLEAR = 'TOGGLE_EDIT_ClEAR';
export const TOGGLE_EDIT_TOUCH = 'TOGGLE_EDIT_TOUCH';
export const TOGGLE_EDIT_UPDATE = 'TOGGLE_EDIT_UPDATE';
export const TOGGLE_EDIT_ADD_STRATEGY = 'TOGGLE_EDIT_ADD_STRATEGY';
export const TOGGLE_EDIT_ADD_GROUP = 'TOGGLE_EDIT_ADD_GROUP';
export const TOGGLE_EDIT_REMOVE_STRATEGY = 'TOGGLE_EDIT_REMOVE_STRATEGY';
export const TOGGLE_EDIT_ADD_STRATEGY_TO_GROUP = 'TOGGLE_EDIT_ADD_STRATEGY_TO_GROUP';
export const TOGGLE_EDIT_REMOVE_STRATEGY_FROM_GROUP = 'TOGGLE_EDIT_REMOVE_STRATEGY_FROM_GROUP';
export const TOGGLE_EDIT_MOVE_STRATEGY = 'TOGGLE_EDIT_MOVE_STRATEGY;';

export function init(featureToggle) {
    debug('init ', featureToggle.name);
    return { type: TOGGLE_EDIT_INIT, featureToggle };
}

export function clearEdit(featureToggleName) {
    debug('clear ', featureToggleName);
    return { type: TOGGLE_EDIT_CLEAR, featureToggleName };
}

export function touch(featureToggleName) {
    debug('touch ', featureToggleName);
    return { type: TOGGLE_EDIT_TOUCH, featureToggleName };
}

export function update(featureToggle) {
    debug('Update ', featureToggle.name);
    return { type: TOGGLE_EDIT_UPDATE, featureToggle };
}

export function addGroup(featureToggleName) {
    debug('Add group ', featureToggleName);
    return {
        type: TOGGLE_EDIT_ADD_GROUP,
        featureToggleName,
    };
}

export function removeStrategy(featureToggleName, index) {
    debug(`Remove strategy ${index} from ${featureToggleName}`);
    return {
        type: TOGGLE_EDIT_REMOVE_STRATEGY,
        featureToggleName,
        index,
    };
}

export function addStrategy(toggleName, strategy) {
    debug(`Add strategy to ${toggleName}`);
    return {
        type: TOGGLE_EDIT_ADD_STRATEGY,
        name: toggleName,
        strategy,
    };
}

export function addStrategyToGroup(featureToggleName, groupIndex, strategy) {
    debug(`Add strategy to group ${groupIndex} for ${featureToggleName}`);
    return {
        type: TOGGLE_EDIT_ADD_STRATEGY_TO_GROUP,
        featureToggleName,
        groupIndex,
        strategy,
    };
}

export function removeStrategyfromGroup(toggleName, groupIndex, id) {
    debug(`Remove strategy from group ${groupIndex} for ${toggleName}`);
    return {
        type: TOGGLE_EDIT_REMOVE_STRATEGY_FROM_GROUP,
        name: toggleName,
        groupIndex,
        id,
    };
}
export function moveStrategy(toggleName, from, to) {
    debug(`Move strategy for ${toggleName}`);
    return {
        type: TOGGLE_EDIT_MOVE_STRATEGY,
        name: toggleName,
        from,
        to,
    };
}
