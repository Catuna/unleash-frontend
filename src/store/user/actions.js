export const USER_UPDATE_USERNAME = 'USER_UPDATE_USERNAME';
export const USER_SAVE = 'USER_SAVE';

export const updateUserName = (value) => ({
    type: USER_UPDATE_USERNAME,
    value,
});

export const save = () => ({
    type: USER_SAVE,
});
