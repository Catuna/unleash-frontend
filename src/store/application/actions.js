import api from '../../data/applications-api';

export const RECEIVE_APPLICATINS = 'RECEIVE_APPLICATINS';
export const ERROR_RECEIVE_APPLICATINS = 'ERROR_RECEIVE_APPLICATINS';

const recieveApplications = (json) => ({
    type: RECEIVE_APPLICATINS,
    value: json,
});

const errorReceiveApplications = (statusCode) => ({
    type: ERROR_RECEIVE_APPLICATINS,
    statusCode,
});

export function fetchApplications () {
    return dispatch => api.fetchAll()
        .then(json => dispatch(recieveApplications(json)))
        .catch(error => dispatch(errorReceiveApplications(error)));
}
