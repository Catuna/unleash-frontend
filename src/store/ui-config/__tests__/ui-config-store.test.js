import { AssertionError } from 'assert';
import reducer from '../index';
import { receiveConfig } from '../actions';

test('should be default state', () => {
    const state = reducer(undefined, {});
    expect(state.toJS()).toMatchSnapshot();
});

test('should be merged state all', () => {
    const uiConfig = {
        headerBackground: 'red',
        slogan: 'hello',
        environment: 'dev',
    };

    const state = reducer(undefined, receiveConfig(uiConfig));
    expect(state.toJS()).toMatchSnapshot();
});

test('should only update headerBackground', () => {
    localStorage.clear();
    const uiConfig = {
        headerBackground: 'black',
    };

    const state = reducer(undefined, receiveConfig(uiConfig));
    expect(state.toJS()).toMatchSnapshot();
});
