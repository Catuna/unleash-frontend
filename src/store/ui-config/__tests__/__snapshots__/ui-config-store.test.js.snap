// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be default state 1`] = `
Object {
  "environment": undefined,
  "headerBackground": undefined,
  "slogan": "A product originally created by FINN.no.",
}
`;

exports[`should be merged state all 1`] = `
Object {
  "environment": "dev",
  "headerBackground": "red",
  "slogan": "hello",
}
`;

exports[`should only update headerBackground 1`] = `
Object {
  "environment": undefined,
  "headerBackground": "black",
  "slogan": "A product originally created by FINN.no.",
}
`;
